test/lib/ansible_test/_internal/containers.py

```diff
@@ -221,6 +235,126 @@ def run_support_container(
     return descriptor
 
 
+def run_container(
+        args: EnvironmentConfig,
+        image: str,
+        name: str,
+        options: t.Optional[list[str]],
+        cmd: t.Optional[list[str]] = None,
+        create_only: bool = False,
+) -> str:
+    """Run a container using the given docker image."""
+    options = list(options or [])
+    cmd = list(cmd or [])
+
+    options.extend(['--name', name])
+
+    network = get_docker_preferred_network_name(args)
+
+    if is_docker_user_defined_network(network):
+        # Only when the network is not the default bridge network.
+        options.extend(['--network', network])
+
+    for _iteration in range(1, 3):
+        try:
+            if create_only:
+                stdout = docker_create(args, image, options, cmd)[0]
+            else:
+                stdout = docker_run(args, image, options, cmd)[0]
+
+            if args.explain:
+                stdout = ''.join(random.choice('0123456789abcdef') for _iteration in range(64))
+
+            return stdout.strip()
+        except SubprocessError as ex:
+            display.error(ex.message)
+            display.warning('Failed to run docker image "{image}". Waiting a few seconds before trying again.')
+            docker_rm(args, name)  # podman doesn't remove containers after create if run fails
+            time.sleep(3)
+
+    raise ApplicationError(f'Failed to run docker image "{image}".')
+
+
+def start_container(args: EnvironmentConfig, container_id: str) -> tuple[t.Optional[str], t.Optional[str]]:
+    """Start a docker container by name or ID."""
+    options: list[str] = []
+
+    for _iteration in range(1, 3):
+        try:
+            return docker_start(args, container_id, options)
+        except SubprocessError as ex:
+            display.error(ex.message)
+            display.warning(f'Failed to start docker container "{container_id}". Waiting a few seconds before trying again.')
+            time.sleep(3)
```

Comment:
I see this mysterious constant of `3` being repeated. Could you move it into a named reusable module-level constant?

Reply:
The 3 second sleep is only used in 2 places in this file. Since it's a somewhat arbitrary sleep, I don't think it warrants a constant for these 2 uses. They just happen to be the same number of seconds.